src/transformers/models/deberta/modeling_deberta.py

```diff
@@ -571,113 +655,194 @@ def forward(
+                )
+            score += p2c_att / scale
+
+        # position->position
+        if "p2p" in self.pos_att_type:
+            pos_query = pos_query_layer[:, :, att_span:, :]
+            p2p_att = torch.matmul(pos_query, pos_key_layer.transpose(-1, -2))
+            p2p_att = p2p_att.expand(query_layer.size()[:2] + p2p_att.size()[2:])
+            if query_layer.size(-2) != key_layer.size(-2):
+                p2p_att = torch.gather(
+                    p2p_att,
+                    dim=-2,
+                    index=pos_index.expand(query_layer.size()[:2] + (pos_index.size(-2), p2p_att.size(-1))),
+                )
+            p2p_att = torch.gather(
+                p2p_att,
+                dim=-1,
+                index=c2p_pos.expand(
+                    [query_layer.size(0), query_layer.size(1), query_layer.size(2), relative_pos.size(-1)]
+                ),
+            )
+            score += p2p_att
 
         return score
 
+    def _pre_load_hook(self, state_dict, prefix, local_metadata, strict, missing_keys, unexpected_keys, error_msgs):
+        self_state = self.state_dict()
+        if ((prefix + "query_proj.weight") not in state_dict) and ((prefix + "in_proj.weight") in state_dict):
+            v1_proj = state_dict[prefix + "in_proj.weight"]
+            v1_proj = v1_proj.unsqueeze(0).reshape(self.num_attention_heads, -1, v1_proj.size(-1))
+            q, k, v = v1_proj.chunk(3, dim=1)
+            state_dict[prefix + "query_proj.weight"] = q.reshape(-1, v1_proj.size(-1))
+            state_dict[prefix + "key_proj.weight"] = k.reshape(-1, v1_proj.size(-1))
+            state_dict[prefix + "key_proj.bias"] = self_state["key_proj.bias"]
+            state_dict[prefix + "value_proj.weight"] = v.reshape(-1, v1_proj.size(-1))
+            v1_query_bias = state_dict[prefix + "q_bias"]
+            state_dict[prefix + "query_proj.bias"] = v1_query_bias
+            v1_value_bias = state_dict[prefix + "v_bias"]
+            state_dict[prefix + "value_proj.bias"] = v1_value_bias
+
+            v1_pos_key_proj = state_dict[prefix + "pos_proj.weight"]
+            state_dict[prefix + "pos_key_proj.weight"] = v1_pos_key_proj
+            v1_pos_query_proj = state_dict[prefix + "pos_q_proj.weight"]
+            state_dict[prefix + "pos_query_proj.weight"] = v1_pos_query_proj
+            v1_pos_query_proj_bias = state_dict[prefix + "pos_q_proj.bias"]
+            state_dict[prefix + "pos_query_proj.bias"] = v1_pos_query_proj_bias
+            state_dict[prefix + "pos_key_proj.bias"] = self_state["pos_key_proj.bias"]
+
+            del state_dict[prefix + "in_proj.weight"]
+            del state_dict[prefix + "q_bias"]
+            del state_dict[prefix + "v_bias"]
+            del state_dict[prefix + "pos_proj.weight"]
+            del state_dict[prefix + "pos_q_proj.weight"]
+            del state_dict[prefix + "pos_q_proj.bias"]
+
```

Comment:
@BigBird01, could you comment on what is this needed for?

Reply:
Sure. In v2 we used a different format to store the attention projection matrix, i.e. q,k,v. In v1, we concatenate them together to benefit from large batch matrix multiplication. However, we found this is not convenient if we want to make some change on that. So in v2 we fall back to the original design that use separate projection matrix for q, k, v. This piece of code is used to convert v1 projection matrix to v2 format.

Reply:
Since we're splitting v1 and v2, there's no need for this anymore, and I believe this method doesn't make comprehension of that model easier. Would you be okay for me to remove this part and update the model weights on the hub? The v1 models will still be able to be loaded in the `DebertaModel`, but not in the `DebertaV2Model`.

Are you okay with that?

Reply:
Yes. We can remove this part of code science we are going to separate the code. The mode has already been updated, so all we need is to update the code and integrate it with master branch.